Allow test success messages to be written to stdout

martanthony opened the pull request:
Closes #18

Macros updated with new parameter, with a default value of `false` to preserve behaviour of existing implementations.

yu-iskw commented:
Thank you for the pull request. But, doesn't the --debug option meet your requirements? It enables us detailed logs on screen.

https://docs.getdbt.com/reference/global-configs/logs#debug-level-logging

martanthony commented:
@yu-iskw I mentioned on the linked issue that `--debug` introduces a lot of noise into the console. This change allows only the 'SUCCESS' message to be printed, instead of every other `debug`-level log event.

yu-iskw commented:
I see. I don't completely understand what you want to get information. Indeed, the changes enable us to show `SUCCESS`. However, we can't get what unit tests are passed. Can you elaborate on the motivateion in detail?

In my case, I use the macro to call all other macros of unit tests. I don't think it gives us meaningful information.

https://github.com/yu-iskw/dbt-unittest/blob/main/integration_tests/macros/tests/test_macros.sql

```
$ cd integration_tests/
$ bash run_unit_tests.sh

14:19:46  Running with dbt=1.7.2
14:19:46  Installing ../
14:19:46  Installed from <local @ ../>
14:19:47  Running with dbt=1.7.2
14:19:48  Registered adapter: bigquery=1.7.0
14:19:48  Found 0 sources, 0 exposures, 0 metrics, 468 macros, 0 groups, 0 semantic models
14:19:48  SUCCESS
14:19:48  SUCCESS
14:19:48  SUCCESS
14:19:48  SUCCESS
14:19:48  SUCCESS
14:19:48  SUCCESS
...
```

martanthony commented:
I saw the same thing as you, then I changed my `run_unit_tests()` macro to be similar to this:

```sql
{% macro run_unit_tests() %}
  {% set unit_tests = ['test_1()',
                       'test_2()',
                       'test_3()',
                       ]
   %}

  {% for unit_test in unit_tests %}
    {% do log("Running " ~ unit_test, info=true) %}
    {% do unit_test %}
  {% endfor %}

{% endmacro %}
```

So that my console output looks like:
```
13:11:33  Running with dbt=1.6.9
13:11:33  Registered adapter: snowflake=1.6.5
13:11:34  Found 593 models, 918 tests, 231 sources, 29 exposures, 0 metrics, 892 macros, 0 groups, 0 semantic models
13:11:34  Running test_1()
13:11:34  Running test_2()
13:11:34  Running test_3()
```

I think it would be more useful to see explicit confirmation that the tests have succeeded, without the console being filled with every single debug-level message:
```
13:11:33  Running with dbt=1.6.9
13:11:33  Registered adapter: snowflake=1.6.5
13:11:34  Found 593 models, 918 tests, 231 sources, 29 exposures, 0 metrics, 892 macros, 0 groups, 0 semantic models
13:11:34  Running test_1()
13:11:34  SUCCESS
13:11:34  Running test_2()
13:11:34  SUCCESS
13:11:34  Running test_3()
13:11:34  SUCCESS
```

I've implemented this so that it's off by default – users who don't want to see `SUCCESS` written to stdout don't have to do anything different in how the implement the macros.

yu-iskw commented:
Does that work in your case? I assume the for loop is stopped if `{% do unit_test %}` is failed. I don't still see strong reasons to support the standard output feature, as we can show full logs using the `--debug` option. And we may have solutions in your case without the changes as I showed. 

```
  {% for unit_test in unit_tests %}
    {% do log("Running " ~ unit_test, info=true) %}
    {% do unit_test %}
    {% do log("Passed " ~ unit_test, info=true) %}
  {% endfor %}
```
